go/enclave/evm/interfaces.go

```diff
@@ -0,0 +1,22 @@
+package evm
+
+import (
+	"context"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	gethcore "github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ten-protocol/go-ten/go/common"
+	"github.com/ten-protocol/go-ten/go/enclave/core"
+)
+
+type EVMFacade interface {
+	ExecuteTx(tx *common.L2PricedTransaction, s *state.StateDB, header *types.Header, gp *gethcore.GasPool, usedGas *uint64, tCount int, noBaseFee bool) *core.TxExecResult
```

Comment:
I still dislike that execute tx is a part of the evm facade; the evm only really cares about calls, the difference is if the resulting state is persisted, tho this will be a bit risky refactor.

Reply:
baby steps.